worlds/stardew_valley/mods/logic/skills_logic.py

```diff
@@ -103,7 +104,13 @@ def can_earn_cooking_skill_level(self, level: int) -> StardewRule:
             return self.logic.cooking.can_cook()
 
     def can_earn_binning_skill_level(self, level: int) -> StardewRule:
-        if level >= 6:
-            return self.logic.has(Machine.recycling_machine)
-        else:
-            return True_()  # You can always earn levels 1-5 with trash cans
+        binning_rule = True_()   # Lets design this around the new craftables to spread out the logic more.
+        if level > 2:
+            binning_rule = binning_rule & self.logic.crafting.can_craft_by_name(ModMachine.trash_bin) & self.logic.crafting.can_craft_by_name(Machine.recycling_machine)
+        if level > 4:
+            binning_rule = binning_rule & self.logic.crafting.can_craft_by_name(ModMachine.composter)
+        if level > 7:
+            binning_rule = binning_rule & self.logic.crafting.can_craft_by_name(ModMachine.recycling_bin)
+        if level > 9:
+            binning_rule = binning_rule & self.logic.crafting.can_craft_by_name(ModMachine.advanced_recycling_machine)
+        return binning_rule
```

Comment:
iirc you can get slightly better performance if you only do AND once at the end, instead of doing `rule = rule & new_rule` many times.

So here you should probably do
```py
# Lets design this around the new craftables to spread out the logic more.
if level <= 2:
    return True_()
binning_rule = [self.logic.has(ModMachine.trash_bin) & self.logic.has(Machine.recycling_machine)]
if level > 4:
    binning_rule.append(self.logic.has(ModMachine.composter))
if level > 7:
    binning_rule.append(self.logic.has(ModMachine.recycling_bin))
if level > 9:
    binning_rule.append(self.logic.has(ModMachine.advanced_recycling_machine))
return binning_rule
```